Adjust the tone of the Panache/Repository guide for JPA and Mongo

InfoSec812 opened the pull request:
Resolves https://github.com/quarkusio/quarkusio.github.io/issues/326

The verbiage in the Repository section of the Panache guide, while explicitly stating "we're not judging", comes across as judgemental/condescending to some people. This is a minor adjustment to the tone of the language used so it can appeal to a broader audience.

InfoSec812 committed:
Adjust the tone of the Panache/Repository guide for JPA and Mongo

machi1990 commented on docs/src/main/asciidoc/mongodb-panache-guide.adoc:
```diff
@@ -425,8 +428,8 @@ public long count(){
 So if Repositories are your thing, you can keep doing them. Even with repositories, you can keep your entities as
-subclasses of `PanacheMongoEntity` in order to get the ID and public fields working, but you can even skip that and
-go back to specifying your ID and using getters and setters if that's your thing. We're not judging.
+subclasses of `PanacheEntity` in order to get the ID and public fields working, but you can even skip that and
```
```suggestion
subclasses of `PanacheMongoEntity ` in order to get the ID and public fields working, but you can even skip that and
```

InfoSec812 replied:
Fixed. Thanks for the review!

loicmathieu commented on docs/src/main/asciidoc/mongodb-panache-guide.adoc:
```diff
@@ -425,8 +428,8 @@ public long count(){
 So if Repositories are your thing, you can keep doing them. Even with repositories, you can keep your entities as
-subclasses of `PanacheMongoEntity` in order to get the ID and public fields working, but you can even skip that and
-go back to specifying your ID and using getters and setters if that's your thing. We're not judging.
+subclasses of `PanacheEntity` in order to get the ID and public fields working, but you can even skip that and
```
subclasses of `PanacheMongoEntity`

loicmathieu commented on docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc:
```diff
@@ -333,8 +333,10 @@ Every query operation accepts passing parameters by index (`Object...`), or by n
+Repositories can be amazing or evil, depending on the complexity of your needs. When they work, 
+they can save you from writing lots of boilerplate and error-prone JPA/SQL/Criteria queries, 
+but when they fail you need a way to drop back to more traditional methods... Don't worry, 
+Panache and Quarkus have you covered either way.
```
The first sentence feels a bit long and still has some judgment. Maybe going straigth to the point should be bettre.

> Repository is a very popular design and can be very accurate for some use case, Panache and Quarkus have you covered ...

InfoSec812 replied:
I like it! Updated.

loicmathieu replied:
Still thinking that the following sentence is not needed:
> but when you need to support a complex use case you can revert
to more traditional methods...

And the Don't worry seems to miss some introduction
> If you need them, don't worry ...

loicmathieu reviewed (changes requested):
One error and maybe the sentence about repository can be simplified

geoand commented:
@InfoSec812 Thanks for the PR! One thing we usually ask contributors to do is squash so a single commit (or into more logical commits if it makes sense) before merging in order to get rid of commits addressing review comments and such.

gsmet commented on docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc:
```diff
@@ -333,8 +333,10 @@ Every query operation accepts passing parameters by index (`Object...`), or by n
-Look, we get it: you have a love/hate relationship with DAOs/Repositories but you can't live without them. We don't judge, we
-know life is tough and we've got you covered.
+Repository is a very popular pattern and can be very accurate for some use case, depending on
+the complexity of your needs. When your use case fits, they can save you from writing lots of boilerplate
```
What is `they`?

gsmet replied:
Reading from the comments, I think the first sentence was modified and the second sentence was not and is definitely not accurate anymore.

gsmet commented on docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc:
```diff
@@ -376,7 +378,7 @@ public long count(){
 So if Repositories are your thing, you can keep doing them. Even with repositories, you can keep your entities as
 subclasses of `PanacheEntity` in order to get the ID and public fields working, but you can even skip that and
-go back to specifying your ID and using getters and setters if that's your thing. We're not judging.
+go back to specifying your ID and using getters and setters if that's your thing. Use what works for you.
```
+1

gsmet commented on docs/src/main/asciidoc/mongodb-panache-guide.adoc:
```diff
@@ -384,7 +384,10 @@ will use the entity field name: `name`, and native query will use the MongoDB fi
 
-Look, we get it: you have a love/hate relationship with DAOs/Repositories but you can't live without them. We don't judge, we know life is tough and we've got you covered.
+Repository is a very popular pattern and can be very accurate for some use case, depending on
+the complexity of your needs. When your use case fits, they can save you from writing lots of boilerplate
```
Same question :).

gsmet reviewed (changes requested):
Can you have a look at my comments? Thanks!

InfoSec812 commented:
> @InfoSec812 Thanks for the PR! One thing we usually ask contributors to do is squash so a single commit (or into more logical commits if it makes sense) before merging in order to get rid of commits addressing review comments and such.

@geoand I've never really developed a habit of using rebase/squash because I prefer the added details in my commit history. Is there a command which you recommend to make it simple?

geoand commented:
@InfoSec812 I usually use IntelliJ's build in UI which makes things super easy. Otherwise git's interactive rebase (as explained [here](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#interactive-rebase) for example) is what I occasionally use

gsmet commented:
I made a few additional adjustments and rebased. I think it's good to go now but will wait for CI.
